core/shared/src/main/scala/scalaz/zio/ZManaged.scala

```diff
@@ -112,14 +119,19 @@ final case class ZManaged[-R, +E, +A](reserve: ZIO[R, E, ZManaged.Reservation[R,
 }
 
 object ZManaged {
-  final case class Reservation[-R, +E, +A](acquire: ZIO[R, E, A], release: ZIO[R, Nothing, _])
 
   /**
    * Lifts a `ZIO[R, E, R]` into `ZManaged[R, E, R]` with a release action.
    */
   final def make[R, E, A](acquire: ZIO[R, E, A])(release: A => ZIO[R, Nothing, _]): ZManaged[R, E, A] =
     ZManaged(acquire.map(r => Reservation(IO.succeed(r), release(r))))
 
+  /**
+   * Lifts a pure `Reservation[R, E, A]` into `ZManaged[R, E, A]`
+   */
+  final def reserve[R, E, A](reservation: Reservation[R, E, A]): ZManaged[R, E, A] =
```

Comment:
Looks great, maybe worth duplicating some Scaladoc here to save people the trouble.